File: code/model/status.py

from .owentity import *
from .user import User
from .wall import Wall
from .graph import Graph
from sparrow import *

class Status(RTOwEntity):
    key = SID = KeyProperty()
    author = RTReference(User)
    wall = RTReference(Wall)
    date = Property(int)
    date_edited = Property(int)  # If they are the same, no edits
    text = Property(str)
    graph = Property(int, required=False)  # This is secretly a reference to Graph
    

File: code/js/prototypes/status.js
Status.prototype = new DataType();
Status.prototype.constructor = Status;
Status.prototype._key = ["SID"];

function Status(SID, date, date_edited, author_UID, wall_WID, text, graph) {
	this.SID = SID;
	this.date = date;
	this.date_edited = date_edited;
	this.author_UID = author_UID;
	this.wall_WID = wall_WID;
	this.text = text
    this.graph = graph;
}
